pkg/controllers/management/globaldns/globaldnsprovider_catalog_launcher.go

```diff
@@ -172,9 +174,14 @@ func (n *ProviderCatalogLauncher) createUpdateExternalDNSApp(obj *v3.GlobalDNSPr
 			Controller: &controller,
 		}}
 
+		creator, err := n.systemAccountManager.GetSystemUser(localClusterName)
+		if err != nil {
+			return nil, err
+		}
+
 		toCreate := pv3.App{
 			ObjectMeta: metav1.ObjectMeta{
-				Annotations:     CopyCreatorID(nil, obj.Annotations),
+				Annotations:     map[string]string{cattleCreatorIDAnnotationKey: creator.Name},
```

Comment:
Do we have to copy any annotations from `obj`? `CopyCreatorID` function kind intended to do that I guess.

Reply:
We just copy the creator id from obj before.